Pythonwin/pywin/framework/startup.py

```diff
@@ -9,8 +9,25 @@
 # this runs OK.  Errors in imported modules are much better - the messages go somewhere (not any more :-)
 
 import sys
+import os
 import win32ui
 
+if not sys.argv:
+	# Initialize sys.argv from commandline. When sys.argv is empty list (
+	# different from [''] meaning "no cmd line arguments" ), then C
+	# bootstrapping or another method of invocation failed to initialize
+	# sys.argv and it will be done here. ( This was a workaround for a bug in
+	# win32ui but is retained for other situations. )
+	import win32api
+	from ctypes import wintypes, windll, c_int, POINTER
+	windll.shell32.CommandLineToArgvW.restype = POINTER(wintypes.LPWSTR)
+	windll.shell32.CommandLineToArgvW.argtypes = (wintypes.LPCWSTR, wintypes.LPINT)
+	argc = c_int() 
+	argv = windll.shell32.CommandLineToArgvW(win32api.GetCommandLine(), argc)
```

Comment:
According to https://docs.microsoft.com/en-us/windows/win32/api/shellapi/nf-shellapi-commandlinetoargvw the functions need a LocalFree call on argv. Is this handled differently in ctypes?

Reply:
Here the `argv` pointer and allocation is simply left as one-time init'ed global module variable pointing to this small per-process heap memory which then dies with the process. It can be still inspected after startup :
```
>>> pywin.framework.startup.argv._b_needsfree_
1
```
.. or I was too lazy to take care of that otherwise in this workaround. It think it does not matter.
(The ctypes `_b_needsfree_` flag likely only applies to the storage space for the pointer itself. ctypes could principally not know automatically about this different type of local heap.)

This 2nd patch now has only little justification left after the 1st, certain odd custom invocations  .. 
But not really a good motive, so I think better drop it on next/final rebase.